watson_developer_cloud/personality_insights_v3.py

```diff
@@ -12,76 +14,813 @@
+        :param str accept_language: The desired language of the response. For two-character arguments, regional variants are treated as their parent language; for example, `en-US` is interpreted as `en`. You can specify any combination of languages for the input and response content.
+        :param bool raw_scores: If `true`, a raw score in addition to a normalized percentile is returned for each characteristic; raw scores are not compared with a sample population. If `false` (the default), only normalized percentiles are returned.
+        :param bool csv_headers: If `true`, column labels are returned with a CSV response; if `false` (the default), they are not. Applies only when the `Accept` header is set to `text/csv`.
+        :param bool consumption_preferences: If `true`, information about consumption preferences is returned with the results; if `false` (the default), the response does not include the information.
+        :return: A `dict` containing the `Profile` response.
+        :rtype: dict
+        """
+        if content is None:
+            raise WatsonInvalidArgument('content must be provided')
+        if content_type is None:
+            raise WatsonInvalidArgument('content_type must be provided')
         headers = {
             'content-type': content_type,
-            'content-language': content_language,
-            'accept': accept,
-            'accept-language': accept_language
+            'Content-Language': content_language,
+            'Accept-Language': accept_language
         }
-
         params = {
             'version': self.version,
             'raw_scores': raw_scores,
-            'consumption_preferences': consumption_preferences,
-            'csv_headers': csv_headers
+            'csv_headers': csv_headers,
+            'consumption_preferences': consumption_preferences
         }
-
-        return self.request(
-            method='POST', url='/v3/profile', data=text, params=params,
+        if content_type == 'application/json' and isinstance(content, dict):
+            data = json.dumps(content)
+        else:
+            data = content
+        response = self.request(
+            method='POST',
+            url='/v3/profile',
             headers=headers,
-            accept_json=(accept is None or accept == 'application/json'))
+            params=params,
+            data=data,
+            accept_json=True)
```

Comment:
Change the generator to as it has to take care of csv:
```
accept_json= csv_headers is None
```

Reply:
Handling for multiple accept types is coming.